docs/dev-sdk.md

```diff
@@ -1,59 +1,199 @@
+
+Therefore, for any new messages, enums, etc, please follow the style guide.
+
+### OpenStorage Style
+
 All SDK APIs and values must satisfy by the following:
 
-* Must be readable
-    * SDK APIs and values must be concrete clear values
-    * `string` values are for ids or strings. They are not meant to marshal information as a `yaml`, create a concrete _message_ instead.
+#### Version
+
+Any changes to the protocol must bump the version by one. On the _master_ branch, the _minor_ number is bumped. On _release_ branches, the _patch_ number is bumped.
+
+#### Readability 
+
+**APIs Must be readable**. SDK APIs and values must be concrete clear values. They are used not just by Portworx, but also non-Portworx developers which do not have an understanding of the internals of the Portworx cluster.
+
+#### `string` values.
+
+* `string` types should be used only for ids, messages, or opaque values. They are not meant to marshal information as a `yaml`. Instead create a concrete _message_.
 * No value options passed as `string`
-    * Instead of passing "Done", or "paused", use enums for these value. Making it clear to the reader.
-* Services:
-    * Services should be in the format `OpenStorage<Type>`.
-    * Note that the service is a collection of APIs and are grouped as such in the documentation.
-        * Here is an example for [OpenStorageClusterService](https://libopenstorage.github.io/w/generated-api.html#serviceopenstorageapiopenstoragecluster)
-* APIs
-    * If it is a new service, then it should have `Create`, `Inspect`, `Delete`, or `Enumerate` style APIs.
-    * All APIs **must** have a single message for the request and a single message for the response with the following style: `Sdk<Service Type><Api Name>Request|Response`
-* Enums
-    * Enum of zero value should be labeled as `XXX_UNKNOWN`, `XXX_NONE`, or `XXX_UNDEFINED` to check if it was not set.
-    * Top level enums must be unique across the entire proto file.
-* Messages
-    * Try not to use `uint64`. Instead try to use signed `int64`. (There is a reason for this which is why CSI changed all uint64s to in64s in version 0.2, but I can find out why.)
-* Documentation
-    * It is imperative that the documentation is correct since it is used to automatically generate the documentation for https://libopenstorage.github.io .The documentation for these values in the proto files can be in Markdown format.
-
-**NOTE** Most importantly is that these APIs _must_ be supported forever. They will almost never be deprecated since at some point we will have many versions of the clients. So please be clear and careful on the API you create.
+    * Instead of passing "Done", or "paused", use enums for these value, making it clear to the reader.
 
-## Creating new API
+#### SDK Services
```

Comment:
Can include a note about updating default users (system.user) with any services that make sense to add when creating a new service.